### PR TITLE
Total revamp of entire library - API v3 and object-oriented data

### PR DESCRIPTION
**Pull request by ErikBoesen**
The two most notable changes:
* Now uses TBA API v3 instead of v2. There's a lot of differences, so I had to change a great deal. See [README](https://github.com/frc1418/tbapy/tree/apiv3#breaking-changes-between-v0x-and-v1x) for specific changes.
* Library now handles data as objects rather than pure JSON/dicts. This is better for a great deal of reasons.

I made a lot of changes even in addition to these crucial two, so It's quite possible I messed something up here, so please look through it carefully before approving.

### CODE REVIEW

File: tbapy/main.py [hunk 1]
```diff
@@ -1,109 +1,343 @@
-import requests
+from requests import get
+from .models import *
```
Comment (Twinters007) [inline tbapy/main.py]: This is also in `__init__.py`. I'm not too familiar with putting stuff there, but is this needed?

File: tbapy/models.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        self.postal_code = json['postal_code']
+        self.rookie_year = json['rookie_year']
+        self.state_prov = json['state_prov']
+        self.team_number = json['team_number']
+        self.website = json['website']
+
+
+class Event:
+    def __init__(self, json):
+        self.json = json
```
Comment (Twinters007) [inline tbapy/models.py]:
```
for key, value in json.items():
    setattr(self, key, value)
```

**Comment by Twinters007** [comment]
This is pretty good. It's so simple I can't see anything major. Any particular parts you were unsure about?

File: tbapy/main.py (hunk 1 above)
Comment (ErikBoesen) [inline tbapy/main.py]: For some reason it does seem to be necessary. Not sure why. But it was giving me an error if I didn't put it in there. On second thought, I don't think I have to put it in `__init__.py`.

File: tbapy/models.py (hunk 2 above)
Comment (ErikBoesen) [inline tbapy/models.py]: Duh.

Comment (ErikBoesen) [inline tbapy/models.py]: I think I'll leave it like this to make it easier when I need to make it more complicated in the future.

**Pull request merged**